model/incoming_webhook.go

```diff
@@ -63,11 +98,201 @@ func IncomingWebhookListFromJson(data io.Reader) []*IncomingWebhook {
 	return o
 }
 
+// TN added, this should be modified
+
+func (o *IncomingWebhook) ResetModels() {
+	o.HmacAlgorithm	= ""
+	o.TimestampModel = StringInterface{}
+	o.HmacModel = StringInterface{}
+	o.SignedContentModel = StringArray{}
+}
+
+func ValidModelSplitter(s *StringInterface) bool {
+	// empty splitBy is valid
+	splitValue, isSet := s.ValidHeaderModel("SplitBy")
+	if ! isSet {
+		delete(*s, "Index")
+		return true
+	}
+
+	idxValue,_ := s.ValidHeaderModel("Index")
+	if len(splitValue) > 0{
+		if ! (len(idxValue) > 0) {
+			return false
+		}
+		intIdx,err:= strconv.ParseInt(idxValue, 10, 0)
+		if err != nil {
+			return false
+		} else if int(intIdx) > 5 || int(intIdx) < 0 {
+			return false
+		}
+	} else {
+		if len(idxValue) > 0 {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (s StringInterface) ValidHeaderModel(mapKey string) (string, bool) {
+	var isSet bool = false
+	var val string = ""
+	if s[mapKey] != nil {
```

Comment:
When you try to access a map in go, it will let you know if the value is set if you put a second value. Therefore this should get simpler if you do something like:
```headerPart, isSet := s[mapKey]```
Is there any reason why you are doing it this way?

Reply:
@larkox No other reason, just didn't remember that probably:) I will change this in next commit. thanks